### PR TITLE
Bitwise Module Revamp

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: content/4_Gold/Intro_Bitwise.mdx [hunk 1]
```diff
@@ -28,10 +29,102 @@ redirects:
...
+
+
+<Spoiler title="Proof">
+Recall a bit in $a \oplus b$ is $1$ only if the bit in $a$ is different from the bit in $b$. By negating $a \& b$, the bits that are left on are in the following format:
+
+- If it is $1$ in $a$ and $0$ in $b$
+- If it is $0$ in $a$ and $1$ in $b$
+- If it is $0$ in $a$ and $0$ in $b$
+
+Now this look's pretty great, but we need to get rid of the third case. By taking the bitwise AND with $a | b$, the ones that are left on is only if there is a $1$ in either $a$ or $b$. Obviously, the third case isnt included in $a | b$ since both bits are off, and we successfully eliminate that case.
```
Comment (SansPapyrus683) [inline content/4_Gold/Intro_Bitwise.mdx:72]: > look's

File: content/4_Gold/Intro_Bitwise.mdx [hunk 2]
```diff
@@ -28,10 +29,102 @@ redirects:
...
+
+<Spoiler title="Hint">
+
+Our queries are quite limited, as we can only find the AND and OR of two numbers. Maybe we can
+combine the usage of these two operations to find the value of an operation that would be more
+useful to us?
+
+</Spoiler>
+
+<Spoiler title="Solution">
```
Comment (SansPapyrus683) [inline content/4_Gold/Intro_Bitwise.mdx:47]: modules don't have spoilers around solutions

File: content/4_Gold/Intro_Bitwise.problems.json [hunk 3]
```diff
@@ -1,31 +1,107 @@
 {
   "MODULE_ID": "intro-bitwise",
+
+  "sample": [
+    {
+      "uniqueId": "cf-1556D",
+      "name": "Take  Guess",
```
Comment (SansPapyrus683) [inline content/4_Gold/Intro_Bitwise.problems.json:7]:
```suggestion
      "name": "Take Guess",
```

File: content/4_Gold/Intro_Bitwise.mdx [hunk 4]
```diff
@@ -28,10 +29,102 @@ redirects:
...
+	for (int i = 4; i <= n; i++) { a.push_back(sum(i - 1, i) - a.back()); }
+	sort(a.begin(), a.end());
+	cout << "finish " << a[k - 1] << endl;
+}
+```
+</CPPSection>
+</LanguageSection>
 
+Now let's take a look on how to implement addition and multiplication using only bitwise operators.
 ## Example - Addition
```
Comment (SansPapyrus683) [inline content/4_Gold/Intro_Bitwise.mdx:128]:
```suggestion
Now let's take a look on how to implement addition and multiplication using only bitwise operators.

## Example - Addition
```

File: content/4_Gold/Intro_Bitwise.mdx [hunk 5]
```diff
@@ -28,10 +29,102 @@ redirects:
...
+</Spoiler>
+
+<Spoiler title="Solution">
+
+In fact, we can figure out the sum of two numbers using just their AND, OR and XOR values! Suppose we know their
+XOR values, we can use the following property:
+
+$a + b = 2 \cdot (a \& b) + a \oplus b$
+
+<Spoiler title="Proof">
```
Comment (SansPapyrus683) [inline content/4_Gold/Intro_Bitwise.mdx:54]: why spoiler the proof

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Intro_Bitwise.mdx (hunk 2 above)
Comment (envyaims) [reply content/4_Gold/Intro_Bitwise.mdx:47]: that ruins the hint though

File: content/4_Gold/Intro_Bitwise.mdx (hunk 5 above)
Comment (envyaims) [reply content/4_Gold/Intro_Bitwise.mdx:54]: idk maybe that want to do it for themselves first

Comment (SansPapyrus683) [reply content/4_Gold/Intro_Bitwise.mdx:54]: add some text in front of it "maybe you wanna do it for yourself, but the proof is here"

Comment (envyaims) [reply content/4_Gold/Intro_Bitwise.mdx:54]: it looks ugly, but if you insist i'll do it

**Review by ryanchou-dev — APPROVED** [review]

File: content/4_Gold/Intro_Bitwise.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply content/4_Gold/Intro_Bitwise.mdx:54]: did you do it

Comment (envyaims) [reply content/4_Gold/Intro_Bitwise.mdx:54]: your mother

File: content/4_Gold/Intro_Bitwise.mdx [hunk 6]
```diff
@@ -28,9 +29,100 @@ redirects:
...
+	// get the rest of the array
+	for (int i = 4; i <= n; i++) { a.push_back(sum(i - 1, i) - a.back()); }
+	sort(a.begin(), a.end());
+	cout << "finish " << a[k - 1] << endl;
+}
+```
+</CPPSection>
+</LanguageSection>
+
+Now let's take a look on how to implement addition and multiplication using only bitwise operators.
```
Comment (SansPapyrus683) [inline content/4_Gold/Intro_Bitwise.mdx:125]: didn't you just explain how to do this above

Comment (envyaims) [reply content/4_Gold/Intro_Bitwise.mdx:125]: no, i still used + when getting the sum

**Pull request merged**